### PR TITLE
feat: address management added

### PR DESCRIPTION
**Pull request by JowelTisso**
This PR is for address management :

### CODE REVIEW

File: src/pages/address/components/address_field/AddressField.jsx [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+            `/api/user/addresses/${selectedAddress._id}`,
+            {
+              address: { ...addressData, _id: selectedAddress._id }, // To make _id consistent through out updates
+            },
+            {
+              headers: { authorization: getUserToken() },
+            }
+          );
+          console.log(res);
+          if (res?.status === 200) {
```
Comment (rahulyadav139) [inline src/pages/address/components/address_field/AddressField.jsx:117]: I don't think we need to use optional chaining here. We send data successfully we always get a response.

File: src/pages/products/Products.jsx [hunk 2]
```diff
@@ -286,13 +286,9 @@ const Products = () => {
         </div>
         <div className="product-content-card-section pd-bottom-4x">
           {productData.map((item) => (
-            <Link
-              to={`/product/${item._id}`}
-              className="no-deco product-card"
-              key={item._id}
-            >
+            <div className="product-card" key={item._id}>
               <ProductCard data={item} navigate={navigate} />
```
Comment (rahulyadav139) [inline src/pages/products/Products.jsx:290]: You don't need to pass navigate function here. Wherever you require it just import it there. It is a context.

**Review by rahulyadav139 — APPROVED** [review]
Other than these suggestions everything is looking great. Great work.

File: src/pages/address/Address.jsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+import { Modal } from "./components/modal/Modal";
+import SavedAddress from "./components/saved_address/SavedAddress";
+import { useAddress } from "./context/address-context";
+import { getUserToken } from "../../utils/TokenHelper";
+
+function Address() {
+  const [isAddingAddress, setIsAddingAddress] = useState(false);
+  const [isModalActive, setIsModalActive] = useState(false);
+  const [isEditActive, setIsEditActive] = useState(false);
+  const [addressId, setAddressId] = useState("");
```
Comment (ivinayakg) [inline src/pages/address/Address.jsx:14]: if the no. of state increase to 3 or more delegate into a reducer

File: src/pages/address/Address.jsx [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  const getSelectedAddressId = (_id) => {
+    setAddressId(_id);
+  };
+
+  const deleteAddressFromServer = async () => {
```
Comment (ivinayakg) [inline src/pages/address/Address.jsx:54]: add the API calls in some other file

**Review by ivinayakg — APPROVED** [review]
code is good, just dont add files which are not required in that particular PR

File: src/pages/address/components/address_field/AddressField.jsx (hunk 1 above)
Comment (JowelTisso) [reply src/pages/address/components/address_field/AddressField.jsx:117]: Ok, got it

File: src/pages/products/Products.jsx (hunk 2 above)
Comment (JowelTisso) [reply src/pages/products/Products.jsx:290]: Ok, will do that

**Review by manismk — APPROVED** [review]
Looks good. Functionality works properly.
Good to see validation for address fields.

File: src/pages/address/Address.jsx (hunk 3 above)
Comment (JowelTisso) [reply src/pages/address/Address.jsx:14]: Ok will try that

**Pull request merged**